### PR TITLE
Add `.appendToArray()` method

### PR DESCRIPTION
**Pull request by subins2000**
* Addresses https://github.com/sindresorhus/electron-store/issues/52
* Addresses #109

Allows appending or creating an array item with `.appendToArray` :

```
store.appendToArray('foo', 'a') // foo will be ['a']
store.appendToArray('foo', 'b') // foo will be ['a', 'b']
```

### CODE REVIEW

File: test/index.ts [hunk 1]
```diff
@@ -99,6 +99,17 @@ test('.set() - invalid key', t => {
 	}, {message: 'Expected `key` to be of type `string` or `object`, got number'});
 });
 
+test('.appendToArray()', t => {
```
Comment (sindresorhus) [inline test/index.ts:102]: This needs an assertion for the case where the value is already set to a different type.

I would also appreciate more assertions if you can try to think of some more.

File: source/index.ts [hunk 2]
```diff
@@ -223,6 +223,25 @@ class Conf<T extends Record<string, any> = Record<string, unknown>> implements I
 		this.store = store;
 	}
 
+	/**
```
Comment (sindresorhus) [inline source/index.ts:226]: The types here should limit it to only allow using `appendToArray` on a key defined to have an array value or unknown.

File: source/index.ts [hunk 3]
```diff
@@ -223,6 +223,25 @@ class Conf<T extends Record<string, any> = Record<string, unknown>> implements I
...
+    Append an item to array
+
+	@param {key|object} - You can use [dot-notation](https://github.com/sindresorhus/dot-prop) in a key to access nested properties. Or a hashmap of items to set at once.
+	@param value - Must be JSON serializable. Trying to set the type `undefined`, `function`, or `symbol` will result in a `TypeError`.
+    */
```
Comment (sindresorhus) [inline source/index.ts:231]:
```suggestion
*/
```

File: source/index.ts [hunk 4]
```diff
@@ -223,6 +223,25 @@ class Conf<T extends Record<string, any> = Record<string, unknown>> implements I
...
+	appendToArray<Key extends keyof T>(key: Key, value: Required<T>[Key]): void;
+	// This overload is used for dot-notation access.
+	appendToArray<Key extends string>(key: Key, value: Required<T>[Key]): void {
+		const list = this.get(key, [] as T[Key]);
+
+		if (!Array.isArray(list)) {
+			throw new TypeError('Config param already set and is not array');
```
Comment (sindresorhus) [inline source/index.ts:238]:
```suggestion
			throw new TypeError(`The key \`${key}\` is already set to a non-array value`);
```

File: source/index.ts [hunk 5]
```diff
@@ -223,6 +223,25 @@ class Conf<T extends Record<string, any> = Record<string, unknown>> implements I
...
+		}
+
+		list.push(value);
+		this.set(key, list);
```
Comment (sindresorhus) [inline source/index.ts:242]:
```suggestion
		this.set(key, [...list, value]);
```

File: source/index.ts [hunk 6]
```diff
@@ -223,6 +223,25 @@ class Conf<T extends Record<string, any> = Record<string, unknown>> implements I
...
+	/**
+    Append an item to array
+
+	@param {key|object} - You can use [dot-notation](https://github.com/sindresorhus/dot-prop) in a key to access nested properties. Or a hashmap of items to set at once.
+	@param value - Must be JSON serializable. Trying to set the type `undefined`, `function`, or `symbol` will result in a `TypeError`.
+    */
+	appendToArray<Key extends keyof T>(key: Key, value: Required<T>[Key]): void;
+	// This overload is used for dot-notation access.
+	appendToArray<Key extends string>(key: Key, value: Required<T>[Key]): void {
+		const list = this.get(key, [] as T[Key]);
```
Comment (sindresorhus) [inline source/index.ts:235]:
```suggestion
		const array = this.get(key, [] as T[Key]);
```

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
It must be documented in the readme.

File: source/index.ts (hunk 2 above)
Comment (subins2000) [reply source/index.ts:226]: I tried many, but can't figure it out. The best I could come up with was
```ts
appendToArray<Key extends keyof T, Index extends keyof any[]>(key: Key, value: T[Key][Index]): void;
```

Comment (subins2000) [reply source/index.ts:226]: @sindresorhus I'm not able to make it, would like to know the solution please

**Comment by LanceGundersen** [comment]
This is exactly what I am looking for! Anything I can help to get this merged?

**Comment by sindresorhus** [comment]
> Anything I can help to get this merged?

Yes, see: https://github.com/sindresorhus/conf/pull/124#discussion_r480275958

**Comment by sindresorhus** [comment]
@subins2000 Bump

**Pull request closed without merging**